presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/StaticMetastoreLocator.java

```diff
@@ -51,6 +59,12 @@ public StaticMetastoreLocator(List<URI> metastoreUris, String metastoreUsername,
                 .map(uri -> HostAndPort.fromParts(uri.getHost(), uri.getPort()))
                 .collect(toList());
         this.metastoreUsername = metastoreUsername;
+        requireNonNull(authenticationType, "authenticationType is null");
```

Comment:
Let's move this validation into the other constructor. It's weird to pass an additional dependency here just for validation. Alternatively, we could do this validation in `ThriftMetastoreModule` as an eager singleton. See `SystemConnectorModule` for an example.